crates/uv-distribution-types/src/dist_error.rs

```diff
@@ -1,8 +1,69 @@
-use std::fmt::{Display, Formatter};
+use std::fmt::{Debug, Display, Formatter};
+use std::ops::Deref;
 use uv_normalize::{ExtraName, GroupName, PackageName};
 use uv_pep440::Version;
 use version_ranges::Ranges;
 
+/// Orphan-rule workaround for inspecting build frontend errors.
+///
+/// Normally we would match on the variants, but we have to pass this error through an opaque type
+/// to avoid cyclical crate dependencies between the build frontend and resolver/installer.
+pub trait IsBuildBackendError: std::error::Error + Send + Sync + 'static {
+    /// Returns whether the build backend failed to build the package, so it's not a uv error.
+    fn is_build_backend_error(&self) -> bool;
+}
+
+/// Wrapper type to make `thiserror`'s `AsDynError` work with `IsBuildFrontendError`.
+///
+/// `thiserror` does not recognize `Box<dyn IsBuildFrontendError + Send + Sync + 'static>` as
+/// error source by itself.
+///
+/// TODO(konsti): The `From<IsBuildBackendError>` blocks implementing `IsBuildBackendError` on the
+/// type itself, so the wrapped
+pub struct AnyErrorBuild(Box<dyn IsBuildBackendError>);
```

Comment:
`AnyBuildError`?

Reply:
It's more like "`anyhow::Error`, but with an extra trait for build error querying on top.